src/pelib/ImageLoader.cpp

```diff
@@ -1761,109 +1757,13 @@ int PeLib::ImageLoader::saveNtHeaders(
 {
 	// Calculate the size of the optional header. Any version of PE file,
 	// 32 or 64-bit, must have this field set to a correct value.
-	std::uint32_t sizeOfOptionalHeader = getFieldOffset(PELIB_MEMBER_TYPE::OPTHDR_sizeof_fixed) + optionalHeader.NumberOfRvaAndSizes * sizeof(PELIB_IMAGE_DATA_DIRECTORY);
```

Comment:
Here, a big chunk of code was removed. What was it doing and why don't we need it anymore?

Reply:
This is because previous version reconstructed NT headers and then saved it to the file. This version saves the NT headers directly from the PE pages as-is.